app/lib/features/chat_ng/widgets/chat_bubble.dart

```diff
@@ -1,40 +1,40 @@
 import 'package:acter/common/themes/acter_theme.dart';
+import 'package:acter/features/chat_ng/models/message_metadata.dart';
+import 'package:acter/features/chat_ng/widgets/reply_preview.dart';
 import 'package:flutter/material.dart';
 import 'package:acter/common/extensions/options.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
 
 class ChatBubble extends StatelessWidget {
   final Widget child;
   final int? messageWidth;
-  final bool wasEdited;
-  final bool isNextMessageInGroup;
+  final MessageMetadata metadata;
   final BoxDecoration decoration;
   final CrossAxisAlignment bubbleAlignment;
 
   // default private constructor
   const ChatBubble._inner({
     super.key,
     required this.child,
-    required this.wasEdited,
+    required this.metadata,
```

Comment:
I feel like the chat-bubble widget is getting too much intelligence and builds up into another mega widget that does a lot of things depending on several inputs...

I think rather than have it pack in a lot information that is unpacked again, I'd rather it to have a `final List<Widgets>? children` and have the outer renderer be _smart_ about what should go in ...